htmd/qm/psi4.py

```diff
@@ -0,0 +1,244 @@
+                raise NotImplementedError
+
+            # Write the molecule
+            f.write('molecule MOL {\n    %d %d\n' % (self._charge, self.multiplicity))
+            elements = self._molecule.element
+            coords = self._molecule.coords[:, :, iframe]
+            for element, coord in zip(elements, coords):
+                f.write('    %-2s %10f %10f %10f\n' % (element, coord[0], coord[1], coord[2]))
+            f.write('    symmetry c1 }\n\n')
+
+            if self._restrained_dihedrals is not None:
+                dihedrals = ['%d %d %d %d' % tuple(dihedral) for dihedral in self._restrained_dihedrals]
+                dihedrals = ', '.join(dihedrals)
+                f.write('set optking { frozen_dihedral = (" %s ") }\n\n' % dihedrals)
+
+            # Enable a dynamic optimization algorithm selection to converge problematic cases:
+            # http://www.psicode.org/psi4manual/master/optking.html#dealing-with-problematic-optimizations
+            if self.optimize:
+                f.write('set optking { dynamic_level = 1 }\n\n')
+
+            theory_correction = 'SCF' if self.theory == 'HF' else self.theory
+            theory_correction += '' if self.correction == 'none' else '-%s' % self.correction
+            function = 'optimize' if self.optimize else 'energy'
+            f.write('energy, wfn = %s(\'%s\', return_wfn=True)\n\n' % (function, theory_correction))
+
+            # Psi4 changes directory then PCM is used, so we need to return
+            f.write('import os\n')
+            f.write('os.chdir(os.environ[\'HTMD_PSI4_WORKDIR\'])\n\n')
+
+            f.write('oeprop(wfn, \'DIPOLE\', \'QUADRUPOLE\', \'MULLIKEN_CHARGES\')\n')
+            if self.esp_points is not None:
+                f.write('oeprop(wfn, \'GRID_ESP\')\n')
+            f.write('\n')
+
+            f.write('with open(\'psi4out.xyz\', \'w\') as f:\n')
+            f.write('    f.write(\'%d \' )\n' % len(coords))
+            f.write('    f.write(\'%.12f\\n\' % energy)\n')
+            f.write('    f.write(MOL.save_string_xyz())\n')
+
+    def _readOutput(self, directory):
+
+        result = QMResult()
+        result.completed = True
+
+        xyzFile = os.path.join(directory, 'psi4out.xyz')
+        if os.path.exists(xyzFile):
+            with open(xyzFile) as f:
+                result.energy = float(f.readline().split()[1])  # Read the 2nd number on the 1st line
+                result.energy *= 627.509469  # Hartree to kcal
+            result.coords = np.loadtxt(xyzFile, skiprows=2, usecols=(1, 2, 3))
+            result.coords = np.atleast_3d(result.coords)  # TODO get rid of this
+        else:
+            result.errored = True
+
+        if self.esp_points is not None:
+            espFile = os.path.join(directory, 'grid_esp.dat')
+            if os.path.exists(espFile):
+                result.esp_points = self.esp_points
+                result.esp_values = np.loadtxt(espFile)/0.529177249  # Bohrs to Angstoms
```

Comment:
again, a standard library value would be great :D

Reply:
Fixed 9f231ac